api/objects.proto

```diff
@@ -48,6 +48,13 @@ message Node {
 
 	// Certificate is the TLS certificate issued for the node, if any.
 	Certificate certificate = 8 [(gogoproto.nullable) = false];
+
+	// Role is the *observed* role for this node. It differs from the
+	// desired role set in Node.Spec.Role because the role here is only
+	// updated after the Raft member list has been reconciled with the
+	// desired role from the spec. Note that this doesn't show whether the
+	// node has obtained a certificate that reflects its current role.
+	NodeRole role = 9;
```

Comment:
My hesitation about this is that the worker is going to act on this field to trigger certificate renewals. "Reconciled role" has meaning to the manager, but not the worker. From the worker's perspective, "Get a new certificate if your role changes" makes sense, but "Get a new certificate if your reconciled role changes" doesn't make as much sense.

What about changing the field under `Spec` to `DesiredRole` and leaving this one as `Role`? I didn't think of that before, but as long as the field number doesn't change, it's fine to rename protobuf fields.